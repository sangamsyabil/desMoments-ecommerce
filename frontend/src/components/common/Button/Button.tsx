import { Button as MuiButton, makeStyles } from '@material-ui/core';
import clsx from 'clsx';
import React from 'react';
import { CustomTheme } from '../../../theme/muiTheme';

export interface ButtonProps {
  onClick?: () => void;
  externalStyle?: string;
  variant?: 'text' | 'outlined' | 'contained' | undefined;
  color?: 'default' | 'inherit' | 'primary' | 'secondary';
  size?: 'large' | 'medium' | 'small';
  disabled?: boolean;
  textTransform?: 'uppercase' | 'lowercase' | 'capitalize';
}

const useStyles = makeStyles((theme: CustomTheme) => ({
  button: {
    textTransform: ({ textTransform }: ButtonProps): any => (textTransform ? textTransform : 'uppercase'),
    '&:hover': {
      backgroundColor: theme.palette.primary.main,
      color: theme.palette.secondary.light,
    },
  },
}));

const Button: React.FC<ButtonProps> = (props) => {
  const classes = useStyles(props);
  const {
    children,
    onClick,
    externalStyle,
    variant = 'contained',
    color = 'primary',
    size = 'medium',
    disabled = false,
  } = props;
  return (
    <MuiButton
      variant={variant}
      size={size}
      color={color}
      onClick={onClick}
      className={clsx([classes.button, externalStyle])}
      disabled={disabled}
    >
      {children}
    </MuiButton>
  );
};

export { Button };
