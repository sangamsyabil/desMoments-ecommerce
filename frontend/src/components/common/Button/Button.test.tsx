import { render } from '@testing-library/react';
import React from 'react';
import { Button, ButtonProps } from './Button';

const mockData: ButtonProps = {
  onClick: jest.fn(),
  variant: 'text',
  color: 'default',
  size: 'large',
  disabled: false,
  textTransform: 'uppercase',
};

test('Button renders correctly', () => {
  const { asFragment } = render(<Button {...mockData}>Mock button</Button>);
  expect(asFragment()).toMatchSnapshot();
});
