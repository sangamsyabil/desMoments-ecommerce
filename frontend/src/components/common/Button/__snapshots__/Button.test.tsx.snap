// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-1 makeStyles-button-2 MuiButton-textSizeLarge MuiButton-sizeLarge"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Mock button
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
